### ISSUE TITLE
Off-by-one error in negative integer divisions

### ISSUE THREAD

**Issue by AlexAltea** [AlexAltea/htypes#4]
Following off-by-one error occurs in negative integer divisions:

```python
>>> int32(-7) // int32(5)
nint(-2)
```

The expected result would be `-1`. Root cause is that in Python `-7 // 5 == -2`.

**Comment by 0x2b3bfa0** [comment]
Seems like `int(x / y)` removes the fractional part by approximation of the result to 0, but `x // y` truncates the number by approximating the result to the lowest (nearest to **-∞**) integer.

**Comment by AlexAltea** [comment]
Yeah, essentially we need:

```python
def op_div(lhs, rhs):
    return int(a / b)
```

and replacing all occurrences of `operator.div` with our custom `op_div`.

**Comment by AlexAltea** [comment]
Fixes are available on [nativetypes-1.0.3](https://pypi.python.org/pypi/nativetypes).

**Comment by 0x2b3bfa0** [comment]
Thank you!
